### PR TITLE
[photon-targeting] Move C++ targeting classes to photon-targetting

### LINKED ISSUE #1004: Move C++ targetting classes into photon-targetting
Currently, PhotonTargetting classes live in photon-lib in C++ while they live in photon-targetting in Java. This should be updated so that the C++ is moved into photon-targetting.

### PR DESCRIPTION
**Pull request by srimanachanta**
closes #1004 

TODO:

- [x] C++ tests
- [x] How to bring this into photon-lib/vendordep
- [x] Publish to Maven
- [x] Add Packet Functionality

### CODE REVIEW

File: photon-targeting/settings.gradle [hunk 1]
```diff
@@ -1 +1,6 @@
-rootProject.name = 'photon-targeting'
```
Comment (mcm001) [inline photon-targeting/settings.gradle:1]: Fairly sure you still need this

File: shared/config.gradle [hunk 2]
```diff
@@ -20,6 +20,8 @@ nativeUtils.wpi.configureDependencies {
 nativeUtils.wpi.addWarnings()
 nativeUtils.wpi.addWarningsAsErrors()
 
+nativeUtils.setSinglePrintPerPlatform()
```
Comment (mcm001) [inline shared/config.gradle:23]: What's this do?

Comment (srimanachanta) [reply shared/config.gradle:23]: No clue, copied it from vendor template, not sure if we need it.

File: shared/setupBuild.gradle [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+                        include "**/*.h"
+                    }
+                }
+            }
+
+            if(project.hasProperty('includePhotonTargeting')) {
+                binaries.all {
+                    lib project: ':photon-targeting', library: 'photontargeting', linkage: 'shared'
+                }
+            }
```
Comment (mcm001) [inline shared/setupBuild.gradle:35]: Can we pull this into photon-lib's build.gradle somehow? this feels like a violation of seperation of concerns. Maybe as something like this?
```
tasks.withType(NativeComponentSpec).each { binaries.all {
                    lib project: ':photon-targeting', library: 'photontargeting', linkage: 'shared'
                }
```

**Review by mcm001 — APPROVED** [review]

**Pull request merged**